unit_test/sparse/Test_Sparse_spmv.hpp

```diff
@@ -159,85 +159,85 @@ void check_spmv_mv(crsMat_t input_mat, x_vector_type x, y_vector_type y, y_vecto
+                       x_vector_type x,
+                       y_vector_type y,
+                       typename y_vector_type::non_const_value_type alpha,
+                       typename y_vector_type::non_const_value_type beta) {
+  using ExecSpace     = typename crsMat_t::execution_space;
+  using scalar_view_t = typename crsMat_t::values_type::non_const_type;
+  using ScalarA       = typename scalar_view_t::value_type;
+  using my_exec_space = Kokkos::RangePolicy<ExecSpace>;
+
+  double eps = std::is_same<ScalarA,float>::value ? 2*1e-3 : 1e-7;
+  size_t nr = input_mat.numRows();
+  y_vector_type expected_y("expected", nr);
+  Kokkos::deep_copy(expected_y, y);
+  Kokkos::fence();
+
+  sequential_spmv(input_mat, x, expected_y, alpha, beta);
+  KokkosSparse::Experimental::spmv_struct("N", stencil_type, structure,
+                                          alpha, input_mat, x, beta, y);
+
+  int num_errors = 0;
+  Kokkos::parallel_reduce("KokkosKernels::UnitTests::spmv_struct",
+                          my_exec_space(0, y.extent(0)),
+                          fSPMV<y_vector_type, y_vector_type, y_vector_type>(expected_y,y,eps),
+                          num_errors);
+  typedef Kokkos::Details::ArithTraits<typename y_vector_type::non_const_value_type> AT;
+  if(num_errors>0) printf("KokkosKernels::UnitTests::spmv_struct: %i errors of %i with params: %d %lf %lf\n",
+                          num_errors, y.extent_int(0), stencil_type, AT::abs(alpha), AT::abs(beta));
+  EXPECT_TRUE(num_errors==0);
+} // check_spmv_struct
+
+template <typename crsMat_t,
+          typename x_vector_type,
+          typename y_vector_type>
+void check_spmv_mv_struct(const crsMat_t input_mat,
+                          const int stencil_type,
+                          const Kokkos::View<typename crsMat_t::non_const_ordinal_type*, Kokkos::HostSpace> structure,
+                          x_vector_type x,
+                          y_vector_type y,
+                          y_vector_type expected_y,
+                          typename y_vector_type::non_const_value_type alpha,
+                          typename y_vector_type::non_const_value_type beta,
+                          int numMV) {
+  typedef typename crsMat_t::execution_space ExecSpace;
+  typedef typename crsMat_t::values_type::non_const_type scalar_view_t;
+  typedef typename scalar_view_t::value_type ScalarA;
+  typedef Kokkos::RangePolicy<ExecSpace> my_exec_space;
+
+  double eps = std::is_same<ScalarA,float>::value?2*1e-3:1e-7;
+  Kokkos::deep_copy(expected_y, y);
+  Kokkos::fence();
+
+  KokkosSparse::Experimental::spmv_struct("N", stencil_type, structure,
+                                          alpha, input_mat, x, beta, y);
+
+  for(int vectorIdx = 0; vectorIdx < numMV; ++vectorIdx) {
+    auto x_i = Kokkos::subview (x, Kokkos::ALL (), vectorIdx);
+    auto y_i = Kokkos::subview (expected_y, Kokkos::ALL (), vectorIdx);
     Kokkos::fence();
```

Comment:
Need a fence here ?

Reply:
I am not 100% sure but this is a correctness check so I did not want to have any mistake in the check that catch my other mistakes x)
The unit-test runs pretty fast but if you want me to test with and without the fence I can do that.

Reply:
I was worried it was masking some other bug where a fence is needed, if not then it is ok.